mmv1/third_party/terraform/services/cloudquotas/data_source_google_cloud_quotas_quota_infos.go

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+package cloudquotas
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+func DataSourceGoogleCloudQuotasQuotaInfos() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGoogleCloudQuotasQuotaInfosRead,
+
+		Schema: map[string]*schema.Schema{
+			"parent": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"service": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"page_token": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"page_size": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "100",
+			},
+			"next_page_token": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"quota_infos": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"service": {
+							Type:     schema.TypeString,
+							Required: true,
```

Comment:
I think these should all be `Computed: true`, and none should be `Required: true`

Reply:
Updated to `Required: true`.